src/konserve/encryptor.cljc

```diff
@@ -12,38 +12,49 @@
   (-serialize [_ bytes write-handlers val]
     (-serialize serializer bytes write-handlers val)))
 
-(defn null-encryptor [_store-key _config]
+(defn null-encryptor [_config]
   (fn [serializer]
     (NullEncryptor. serializer)))
 
 ;; Following advise at
 ;; https://crypto.stackexchange.com/questions/84439/is-it-dangerous-to-encrypt-lots-of-small-files-with-the-same-key
 
-(defn store-key->iv [store-key]
-  (subvec (vec (edn-hash store-key)) 0 16))
+(defn get-iv [salt key]
+  (subvec (vec (edn-hash ["initial-value" salt key])) 0 16))
 
-(defrecord AESEncryptor [serializer store-key key]
+(defn get-key [salt key]
+  ["key" salt key])
+
+;; TODO cljs support incomplete, prepending of salt is missing
+(defrecord AESEncryptor [serializer key]
   PStoreSerializer
   (-deserialize [_ read-handlers bytes]
-    (let [decrypted (decrypt [store-key key] #?(:clj (.readAllBytes ^ByteArrayInputStream bytes) :cljs bytes)
-                             :iv (store-key->iv store-key))]
+    (let [salt-array (byte-array 64)
+          _ (.read ^ByteArrayInputStream bytes salt-array)
+          salt (map int salt-array)
+          decrypted (decrypt (get-key salt key)
+                             #?(:clj (.readAllBytes ^ByteArrayInputStream bytes) :cljs bytes)
+                             :iv (get-iv salt key))]
```

Comment:
I can guess what iv stands for from the context, but I shouldn't have to guess.

The field `:iv` as well and maybe even the function `get-iv` - even though it is short - should either be renamed or there should be documentation about it.

Reply:
You need to know AES to touch this code and then IV is clear. This is why I put the link to Stackoverflow there, there is no way to write Crypto code without reading about the underlying libraries and algorithms unfortunately. The same holds for hasch, it is tricky to get these right.

Reply:
I have changed the getter though, which helps readability.

Reply:
It would be nice to also take a look at geheimnis and have it as a new dependency on the radar. It is a bit old dependency with experimental RSA code (which probably should not be used and replaced by quantum safe elliptic curve crypto nowadays anyway), but I think its AES wrapper is good.